### ISSUE TITLE
cargo new and open

### ISSUE THREAD

**Issue by c02y** [kwrooijen/cargo.el#104]
When I'm using `cargo-process-new` to create a new bin or lib, how can I make the `cargo-process-new` open the main.rs/lib.rs file automatically after creating the bin/lib?

**Comment by kwrooijen** [comment]
Currently that's not possible. What we could do is add a flag / defcustom to open the `lib.rs` file after the process is finished. This should be relatively easy to do

Edit: Would probably have to add it as an afterhook. Probably extend the `cargo-process--finished-sentinel` function

**Comment by kwrooijen** [comment]
@c02y added in https://github.com/kwrooijen/cargo.el/commit/f777da4463045076fb532bf8f28323f75c9f7d5e

Put this in your configuration `(setq cargo-process--open-file-after-new t)`. Now when you create a new project it'll open the rs file automatically

**Comment by c02y** [comment]
Hi, every time I `cargo-process-new` a new bin/lib, emacs will open a new window called `*Cargo New*` with message like:
`# message1:`
~~~
-*- mode: cargo-process; default-directory: "~/" -*-
Cargo-Process started at Wed Aug 26 22:41:16

/home/chz/.cargo/bin/cargo new ~/rust_projects/closures2 --bin  
[0m[0m[1m[32m     Created[0m binary (application) `/home/chz/rust_projects/closures2` package
~~~

`message2:`
~~~
-*- mode: cargo-process; default-directory: "~/rust_projects/closures2/" -*-
Cargo-Process started at Wed Aug 26 22:44:14

/home/chz/.cargo/bin/cargo new ../closures3 --bin  
[0m[0m[1m[32m     Created[0m binary (application) `../closures3` package

~~~

The first call of `cargo-process-new` will successfully open the main.rs/lib.rs, but when I'm already inside a rust project such as inside main.rs and call `cargo-process-new`, it will create a new project, the project directory is correctly created in my disk, but the new main.rs/lib.rs opened is empty and its path in Emacs is wrong.

For instance:

0. open new Emacs
1. `cargo-process-new` -> ~/rust_projects/closures2 -> bin
2.  a new window `*Cargo New*` is opened with above message(mesage1)
3. the main.rs is opened in a window successfully with the right content (the "hello world" stuff)
4. inside this main.rs, call `cargo-process-new` ->  ../closures3 -> bin
5. the window `*Cargo New*` window contains new message(message2)
6. the main.rs is opened but the buffer is empty
7. the closures3 directory is created in the same directory as closures2 which is `~/rust_projects`
8. but the path of main.rs file for closures3 project in Emacs is `~/rust_projects/closures2/closures3/src/main.rs` which the real location is `~/rust_projects/closures3` in my disk

so after updating the cargo package to cargo-20200825.1811, there are the following 3 issues:
1. it opens the `*Cargo New*` window with unnecessary message, I'm not sure it is error or not since it contains weird characters
2. step6, the content of the opened buffer is empty
4. step8, the path of the new project in Emacs is wrong and different from the new created path in my disk

**Comment by kwrooijen** [comment]
So the problem is that all cargo commands are run in the root directory (the directory that has a Cargo.toml). If that doesn't exist, the current directory. In your case you're doing `../closures3`, which is 1 level up the `Cargo.toml` direcotry. Instead you'd need to do `src/closures3`.

The second problem is that the after hook of cargo-new didn't use the correct directory. Which is fixed in the latest commit (which I just pushed).

So basically: Get the newest cargo.el version. And use paths relative to the current project.

**Comment by c02y** [comment]
Before updating to cargo-20200825.1811, when I'm in main.rs of one project, I always use `../project_name` to create a new project, and it does created the new project_name in the right path in my disk, just like when using cargo-20200825.1811. The thing, I have to open the main.rs file manually, so I posted this feature request.

The problem is when using cargo-20200825.1811, once it opens the new main.rs, the path of the main.rs for the new project is wrong, the buffer is empty and the path doesn't exist in my disk. I'll try the new version and show the result.

**Comment by kwrooijen** [comment]
Right, the creation hasn't changed. The newest commit on GH should fix your current issue regarding Emacs opening the proper buffer. It might not be on Melpa yet though.

**Comment by c02y** [comment]
Thanks, I downloaded the cargo-process.el file and got the old one replaced manually. It did solve all the issues.

BTW: 
I just noticed the old `NOTICE` part in README.md, sorry that I posted a new feature request, and thank you again for adding this feature and solving these issues so quickly.

**Comment by kwrooijen** [comment]
Good to hear! And don't worry about it. I'll gladly help but I can't give feedback on actual _cargo_ specific features.
